### PR TITLE
Extend Explorer RPC api for accumulative charts

### LINKED ISSUE #181: Explorer RPC API extends
It is necessary to implement methods for calculating the number of accounts created in the context of days/hours

### PR DESCRIPTION
**Pull request by andyoknen**
Two methods have been added to calculate the number of accounts created in the context of days/hours:
- getstatisticcontentbyhours
- getstatisticcontentbydays

### CODE REVIEW

File: src/pocketdb/web/PocketExplorerRpc.cpp [hunk 1]
```diff
@@ -52,15 +52,44 @@ namespace PocketWeb::PocketWebRpc
...
         if (request.fHelp)
             throw std::runtime_error(
-                "getstatisticcontent\n"
-                "\nGet statistics for content transactions\n"
+                "getstatisticcontentbyhours\n"
+                "\nGet statistics for content transactions grouped by hours\n"
             );
 
-        return request.DbConnection()->ExplorerRepoInst->GetContentStatistic();
+        int topHeight = chainActive.Height() / 10 * 10;
```
Comment (lostystyg) [inline src/pocketdb/web/PocketExplorerRpc.cpp:63]: What is purpose for "/ 10 * 10" ?

File: src/pocketdb/repositories/web/ExplorerRepository.cpp [hunk 2]
```diff
@@ -121,6 +121,98 @@ namespace PocketDb {
...
+                    select
+                      count()
+                    from Transactions u1 indexed by Transactions_Type_Last_Height_Id
+                    where u1.Type in (100)
+                    and u1.Height <= u.Height
+                    and u1.Last = 1
+                  )cnt
+                from Transactions u indexed by Transactions_Type_HeightByHour
+                where u.Type in (3)
+                  and (u.Height / 60) <= (? / 60)
```
Comment (lostystyg) [inline src/pocketdb/repositories/web/ExplorerRepository.cpp:142]: I can't get how it mixes here  depth that seems to represent time and height.

File: src/pocketdb/web/PocketExplorerRpc.cpp [hunk 3]
```diff
@@ -52,15 +52,44 @@ namespace PocketWeb::PocketWebRpc
...
+            );
+
+        int topHeight = chainActive.Height() / 10 * 10;
+        if (request.params[0].isNum())
+            topHeight = std::min(request.params[0].get_int(), topHeight);
+
+        int depth = 30;
+        if (request.params[1].isNum())
+            depth = std::min(request.params[1].get_int(), depth);
+        depth = depth * 24 * 60;
```
Comment (lostystyg) [inline src/pocketdb/web/PocketExplorerRpc.cpp:90]: Multiply depth here, but divide it by same values in every place we use it after in "ExplorerRepoInst->GetContentStatisticByDays()". Does it make sense? Same is for "GetStatisticContentByHours()"

File: src/pocketdb/web/PocketExplorerRpc.cpp (hunk 1 above)
Comment (andyoknen) [reply src/pocketdb/web/PocketExplorerRpc.cpp:63]: Rounding) A mini hack that emulates the cache)

File: src/pocketdb/repositories/web/ExplorerRepository.cpp (hunk 2 above)
Comment (andyoknen) [reply src/pocketdb/repositories/web/ExplorerRepository.cpp:142]: Blocks are generated once a minute, +- several minutes. Therefore, we take the height as the number of minutes - the error in this case (the explorer graph) is not very important

File: src/pocketdb/web/PocketExplorerRpc.cpp (hunk 3 above)
Comment (andyoknen) [reply src/pocketdb/web/PocketExplorerRpc.cpp:90]: If I understood the question correctly, the essence of division is that in these queries we rely on the calculated index (see migration), so in queries we have to divide to match the index

**Review by lostystyg — APPROVED** [review]

**Review by tawmaz — APPROVED** [review]

**Pull request merged**